frame/bags-list/src/mock.rs

```diff
@@ -0,0 +1,170 @@
+	type Origin = Origin;
+	type Index = u64;
+	type BlockNumber = u64;
+	type Call = Call;
+	type Hash = sp_core::H256;
+	type Hashing = sp_runtime::traits::BlakeTwo256;
+	type AccountId = AccountId;
+	type Lookup = sp_runtime::traits::IdentityLookup<Self::AccountId>;
+	type Header = sp_runtime::testing::Header;
+	type Event = Event;
+	type BlockHashCount = ();
+	type DbWeight = ();
+	type BlockLength = ();
+	type BlockWeights = ();
+	type Version = ();
+	type PalletInfo = PalletInfo;
+	type AccountData = pallet_balances::AccountData<Balance>;
+	type OnNewAccount = ();
+	type OnKilledAccount = ();
+	type SystemWeightInfo = ();
+	type OnSetCode = ();
+}
+
+parameter_types! {
+	pub static BagThresholds: &'static [VoteWeight] = &[10, 20, 30, 40, 50, 60, 1_000, 2_000, 10_000];
+}
+
+impl crate::Config for Runtime {
+	type Event = Event;
+	type WeightInfo = ();
+	type BagThresholds = BagThresholds;
+	type VoteWeightProvider = StakingMock;
+}
+
+type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<Runtime>;
+type Block = frame_system::mocking::MockBlock<Runtime>;
+frame_support::construct_runtime!(
+	pub enum Runtime where
+		Block = Block,
+		NodeBlock = Block,
+		UncheckedExtrinsic = UncheckedExtrinsic,
+	{
+		System: frame_system::{Pallet, Call, Storage, Event<T>, Config},
+		BagsList: crate::{Pallet, Call, Storage, Event<T>},
+	}
+);
+
+pub(crate) mod ext_builder {
+	use super::*;
+
+	/// Default AccountIds and their weights.
+	pub(crate) const GENESIS_IDS: [(AccountId, VoteWeight); 4] =
+		[(1, 10), (2, 1_000), (3, 1_000), (4, 1_000)];
+
+	#[derive(Default)]
+	pub(crate) struct ExtBuilder {
+		ids: Vec<(AccountId, VoteWeight)>,
+	}
+
+	impl ExtBuilder {
+		/// Add some AccountIds to insert into `List`.
+		pub(crate) fn add_ids(mut self, ids: Vec<(AccountId, VoteWeight)>) -> Self {
+			self.ids = ids;
+			self
+		}
+
+		pub(crate) fn build(self) -> sp_io::TestExternalities {
+			sp_tracing::try_init_simple();
+			let storage =
+				frame_system::GenesisConfig::default().build_storage::<Runtime>().unwrap();
+
+			let mut ext = sp_io::TestExternalities::from(storage);
+			ext.execute_with(|| {
+				for (id, weight) in GENESIS_IDS.iter().chain(self.ids.iter()) {
+					frame_support::assert_ok!(List::<Runtime>::insert(*id, *weight));
+				}
+			});
+
+			ext
+		}
+
+		pub(crate) fn build_and_execute(self, test: impl FnOnce() -> ()) {
+			self.build().execute_with(|| {
+				test();
+				List::<Runtime>::sanity_check().expect("Sanity check post condition failed")
+			})
+		}
+
+		pub(crate) fn build_and_execute_no_post_check(self, test: impl FnOnce() -> ()) {
+			self.build().execute_with(test)
+		}
+	}
+}
+
+pub(crate) mod test_utils {
+	use super::*;
+	use list::Bag;
+
+	/// Returns the nodes of all non-empty bags.
+	pub(crate) fn get_bags() -> Vec<(VoteWeight, Vec<AccountId>)> {
+		BagThresholds::get()
+			.into_iter()
+			.chain(std::iter::once(&VoteWeight::MAX)) // assumes this is not an explicit threshold
```

Comment:
needs a small fix here

Reply:
also, it seems to be repeated here and in the benchmarks. Can move into the pallet, givne that you properly gate it with features.

Reply:
97459c50808bd097ff895e2002d36518c8ca1484